larsim/EventGenerator/SingleGen_module.cc

```diff
@@ -447,11 +447,20 @@ namespace evgen {
     , fHistFileName(config().HistogramFile())
     , fPHist(config().PHist())
     , fThetaXzYzHist(config().ThetaXzYzHist())
-    , fEngine(createEngine(0))
+    , fEngine(art::ServiceHandle<rndm::NuRandomService>()->registerAndSeedEngine(createEngine(0),
+                                                                                 "HepJamesRandom",
+                                                                                 "Seed"))
   {
     setup();
     rndm::NuRandomService::seed_t seed;
-    if (config().Seed(seed)) { fEngine.setSeed(seed, 0 /* dummy? */); }
+    if (config().Seed(seed)) {
+      fEngine.setSeed(seed, 0 /* dummy? */);
+      mf::LogInfo("SingleGen") << "Seed set to " << seed << " (from configuration)";
+    }
+    else {
+      seed = fEngine.getSeed();
+      mf::LogInfo("SingleGen") << "Seed set to " << seed;
+    }
```

Comment:
@mantheys 
Can you confirm what printout you now see in, for example, a DUNE far detector particle gun job?

Reply:
Sure! This is the full output for a single electron production in DUNE larsoft (https://github.com/DUNE/dunesw/blob/develop/fcl/dunefd/gen/solar/prodelectron_solar_flat_dune10kt_1x2x6.fcl). The relevant part of the output start at line 65.
```
%MSG-i RANDOM:  SingleGen:generator@Construction 12-Dec-2024 12:07:01 CET  ModuleConstruction
Instantiated HepJamesRandom engine "generator:0:" with seed 0.
%MSG
%MSG-i NuRandomService:  SingleGen:generator@Construction  12-Dec-2024 12:07:01 CET ModuleConstruction
Seeding HepJamesRandom engine "generator:Seed" with seed 575704593.
%MSG
%MSG-i SingleGen:  SingleGen:generator@Construction 12-Dec-2024 12:07:01 CET  ModuleConstruction
Seed set to 575704593
```
[SinglesGen.txt](https://github.com/user-attachments/files/18110485/SinglesGen.txt)

Reply:
Thanks for this Sergio.  So it looks like the code is now initialising a sensible seed when not using the config.
Can you confirm that you get different seeds between two jobs?

Reply:
Yes exactly! That's the behavior I see now.

Reply:
Perfect, thank you!